src/backend/cuda/kernel/ireduce.hpp

```diff
@@ -36,13 +39,17 @@ namespace kernel
         __host__ __device__ MinMaxOp(T val, uint idx) :
             m_val(val), m_idx(idx)
         {
+            if (isNan(val)) {
+                Binary<T, op> ireduce;
+                m_val = ireduce.init();
```

Comment:
I wonder if you could make `init` a static function. That way you don't have to create a binary object. It doesn't really make sense to create a Binary object here.

Reply:
Yes, it works. I have the changes ready. @mark-poscablo I can send you them as a patch file

Reply:
Good idea - I applied the patch from @9prady9, thanks!